src/events/guild/member/guildMemberAdd.ts

```diff
@@ -24,6 +28,33 @@ export default class extends Event {
 			}),
 		);
 
+		const automod: AutoModConfiguration = await databaseManager.getAutoModConfig(member.guild.id);
+		if (automod.avatarHashes.length > 0 && automod.punishmentAvatarBans) {
+			const buffer = await axios.get(member?.displayAvatarURL()!, {
+				responseType: "arraybuffer",
+			});
+			const avatarHash = await imageHash.hash(buffer.data, 8);
+			if (automod.avatarHashes.includes(avatarHash)) {
+				const automodManager: AutoModManager = new AutoModManager();
+
+				let punishment: string = await automodManager.resolveActionWithoutContext(
```

Comment:
```suggestion
				const punishment: string = await automodManager.resolveActionWithoutContext(
```
- https://discord.com/channels/742094927403679816/823636771702308885/942015093292556359 😄